### DISCUSSION TITLE
Firefox Extension that auto detects RSS Links for Subscribing

### DISCUSSION THREAD

**Discussion by monksy** [FreshRSS/FreshRSS#5598] [Q&A]
**Is your feature request related to a problem? Please describe.**

I'm frustrated when it comes to adding an RSS feed from a web page to my instance. The problem is that the RSS link isn't always visible, or sometimes it's hidden in the page it's self's links. I want a plugin that would change colors when an RSS feed is available in the website, and when pressed it would show you the RSS feeds linked to on the page + allow for you to click to add it to your configured instance. 

**Describe the solution you’d like**

I would like for an extension that monitors pages in Firefox for links to RSS feeds, and presents the user with an option if it's present and to automatically add it to your instance. This would save me time from searching or the RSS url or links.

**Describe alternatives you’ve considered**

I've looked at the firefox extensions that are currently in the firefox extension gallery. They only off the ability to monitor your instance. Not to automate the addition of feeds.

**Comment by Frenzie** [comment]
I'm not entirely sure if I understand. Is there something about https://addons.mozilla.org/en-US/firefox/addon/feed-indicator/ and similar you're not happy with?

**Comment by Alkarex** [comment]
I personally use https://addons.mozilla.org/firefox/addon/want-my-rss/ and I am happy with it.
https://github.com/Reeywhaar/want-my-rss

**Comment by daffydock** [comment]
On the FF extension site you can find a number of add-ons that may help you. @Alkarex choice is a great one. Personally, I used to use this one: https://addons.mozilla.org/en-US/firefox/addon/get-rss-feed-url/

There are other options available that may better fit your needs. 
I guess it would be great if a something was part of FreshRSS but realistically, there are already a number of existing add-ons so I do not see the reason why anyone needs to reinvent the wheel on this one. I used the above without problems. 

Here is a link to the larger listing, so you can pick which one better fits your needs:

https://addons.mozilla.org/en-US/firefox/search/?q=rss

**Reply by monksy** [reply]
That does help. however, that's just the first half of it. I'm looking for something that'll find, alert the user in a non-obtrusive way, (Hey I see you're looking at the Malort Blog, this has 2 feeds on it [with the name]) you click to subscribe against your preconfigured instance and it adds it to your subscriptions)

**Reply by daffydock** [reply]
Fair. The one add-on I used to use puts a numbered badge on its icon to let you know that there are RSS feeds in the site and how many. You click on the badge and will show you exactly the ones available. Maybe that is close to what you want? Take a look at the photos included on the add-on's page, where you can copy the RSS feed. 

As far as adding them, you would still have to copy/paste the address via your regular method via the FreshRSS prompts on your instance. 

Although I follow hundreds of feeds now, in the end, is not like I am adding them daily. So, the extra 15-20 seconds it takes to add one as needed is not an annoying task by any means. Albeit I guess your mileage may vary as per your use patterns.    

If you want something more akin to a push-a-button-to-add-to-my-instance solution, then there is this:
https://addons.mozilla.org/en-US/firefox/addon/rsshub-radar/

It was the only one I found back in the day. Didn't use it for two reasons, one it was only available in Chinese --albeit I believe that was addressed last year-- and also, I am wary of putting my server info into an extension I have no, or any idea if I can trust. But again, you might be okay with that.

**Comment by math-GH** [comment]
Have you checked the subscription tools already?
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/938940ff-b8af-448f-a6d2-7f3075cec95b)

**Reply by monksy** [reply]
That's close to what I'm looking for. I'm looking for something that allows you to detect the feed and gives you the option to subscrbe all in one go. 

My current use case is that you find a blog you're interested in.. say oh great scala tricks topics etc... 

https://blog.rockthejvm.com/scala-syntax-tricks-for-expressiveness/

But your next question is: does this have an rss feed? No visible orange button on the page. Well now I need a plugin or to examine the source.

Feed indicator is going to kick you to another page 

moz-extension://372d0caf-3d65-47d9-b3d4-f253168063f0/preview.html?u=https%3A%2F%2Fblog.rockthejvm.com%2Ffeed.xml

And show the URL on the page.. Now I have to click on that.

https://blog.rockthejvm.com/feed.xml

Finally I have the page that I can use that bookmarklet. 

(Note this bookmarklet is a much better experience than clicking on the + beside subscription management.)

It doesn't happen often.. however.. it does demotivate the the experience that I have from going from a website I'm enthusastic about to keeping track of updates via FeedRss.
